src/main/scala/DateParserApp.scala

```diff
@@ -0,0 +1,57 @@
+import wvlet.log.{LogLevel, LogSupport, Logger}
+
+import java.text.{ParseException, SimpleDateFormat}
+import java.util.Date
+import scala.util.Try
+
+
+/** Collection of utils to parse the date.
+ *
+ * @param log_level set logger level
+ * */
+class DateParserApp(log_level: LogLevel = LogLevel.INFO) extends LogSupport {
+
+  // Prepare logger
+  Logger.init
+  Logger.setDefaultLogLevel(log_level)
+  info(s"Logger set to ${Logger.getDefaultLogLevel} level")
+
+  // set some constants
+  val POSSIBLE_DATA_FMTS: List[String] = List("yyyy-MM-dd", "yyyy/MM/dd", "yyyy.MM.dd", "dd-MM-yyyy", "dd/MM/yyyy", "dd.MM.yyyy")
+
+  /** Parse the date with the given format.
+   *
+   * @param fmt  i.e. "dd-MM-yyyy"
+   * @param date i.e. "20-12-2020"
+   * */
+  def parse_date(fmt: String, date: String): Date = {
+    try {
+      val date_formatter = new SimpleDateFormat(fmt)
```

Comment:
Use `java.time.format.DateTimeFormatter` instead of `SimpleDateFormat`. SimpleDateFormat is an old and outdated class, in addition, it has concurrency issues. Java Time library is much better,